airbyte-integrations/connectors/source-weatherstack/source_weatherstack/source.py

```diff
@@ -0,0 +1,214 @@
+
+    def request_params(
+            self,
+            stream_state: Mapping[str, Any],
+            stream_slice: Mapping[str, Any] = None,
+            next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+        # The api requires that we include api_key as a query param so we do that in this method
+        return {'access_key': self.access_key , 'query' : self.query, 'historical_date' : self.historical_date }
+
+    def parse_response(
+            self,
+            response: requests.Response,
+            stream_state: Mapping[str, Any],
+            stream_slice: Mapping[str, Any] = None,
+            next_page_token: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping]:
+        # The response is a simple JSON whose schema matches our stream's schema exactly, 
+        # so we just return a list containing the response
+        return [response.json()]
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        # The API does not offer pagination,
+        # so we return None to indicate there are no more pages in the response
+        return None
+
+class LocationLookup(HttpStream):
+    url_base = "http://api.weatherstack.com/"
+
+    # Set this as a noop.
+    primary_key = None
+
+    def __init__(self, config: Mapping[str, Any], **kwargs):
+        super().__init__()
+        self.query = config['query']
+        self.access_key = config['access_key']
+
+    def path(
+        self, 
+        stream_state: Mapping[str, Any] = None, 
+        stream_slice: Mapping[str, Any] = None, 
+        next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "autocomplete"  
+
+    def request_params(
+            self,
+            stream_state: Mapping[str, Any],
+            stream_slice: Mapping[str, Any] = None,
+            next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+        # The api requires that we include api_key as a query param so we do that in this method
+        return {'access_key': self.access_key , 'query' : self.query }
+
+    def parse_response(
+            self,
+            response: requests.Response,
+            stream_state: Mapping[str, Any],
+            stream_slice: Mapping[str, Any] = None,
+            next_page_token: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping]:
+        # The response is a simple JSON whose schema matches our stream's schema exactly, 
+        # so we just return a list containing the response
+        return [response.json()]
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        # The API does not offer pagination,
+        # so we return None to indicate there are no more pages in the response
+        return None
+
+# Source
+class SourceWeatherstack(AbstractSource):
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        accepted_cities = {"New York", "London"}
```

Comment:
I believe the invalid_config.json test is supposed to fail and your suppose to provide an invalid name?

I do need to rewrite the check connection.

Reply:
I've seen different approaches to making the check_connection. I chose to utilize the python requests in the file to make the API call. I thought that this would be best suited, because this source is just making requests to an api. Also, the queries to the API can be innumerable : A city, zip code, Latitude/Longitde, and multiple cities together in batch for premium API plan.